### PR TITLE
Added QA activities

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Added activities to the QA app. related to verifying restricted route sections and dynamically updating traffic on the route line.

### CODE REVIEW

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RouteRestrictionsActivity.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import com.mapbox.navigation.qa_test_app.utils.Utils
+import com.mapbox.navigation.qa_test_app.utils.Utils.getRouteOriginPoint
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
+import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLine
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineColorResources
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineResources
+
+class RouteRestrictionsActivity : AppCompatActivity() {
```
Comment (Guardiola31337) [inline qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RouteRestrictionsActivity.kt:20]: Run into the following crash at startup

```
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err: java.lang.IllegalArgumentException: Provided serialized route contains RouteOptions. Use DirectionsRoute#fromJson(json, accessToken) instead.
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.mapbox.api.directions.v5.models.DirectionsRoute.fromJson(DirectionsRoute.java:184)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.mapbox.navigation.qa_test_app.view.RouteRestrictionsActivity.getRoute(RouteRestrictionsActivity.kt:96)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.mapbox.navigation.qa_test_app.view.RouteRestrictionsActivity.initStyle$lambda-1(RouteRestrictionsActivity.kt:83)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.mapbox.navigation.qa_test_app.view.RouteRestrictionsActivity.lambda$zyVQH6eGWFNs3Hi7iD5uctqcRiU(Unknown Source:0)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.mapbox.navigation.qa_test_app.view.-$$Lambda$RouteRestrictionsActivity$zyVQH6eGWFNs3Hi7iD5uctqcRiU.onStyleLoaded(Unknown Source:2)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.mapbox.maps.StyleObserver.onStyleLoaded(StyleObserver.kt:55)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.mapbox.maps.NativeObserver.notify(NativeObserver.kt:57)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at android.os.MessageQueue.nativePollOnce(Native Method)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at android.os.MessageQueue.next(MessageQueue.java:335)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at android.os.Looper.loop(Looper.java:183)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:7664)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
2021-08-17 13:00:09.366 16921-16921/com.mapbox.navigation.qa_test_app W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Review by Guardiola31337 — COMMENTED** [review]
Left a couple of comments @cafesilencio

**Comment by cafesilencio** [comment]
The activity for testing route restrictions is broken and will need updating. There are changes to the restricted road implementation in progress. When these changes have landed the activity will be updated.  The issue ticket is here: https://github.com/mapbox/mapbox-navigation-android/issues/4723

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/InactiveRouteStylingActivity.kt [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+            val routeOrigin = Utils.getRouteOriginPoint(route)
+            val cameraOptions = CameraOptions.Builder().center(routeOrigin).zoom(14.0).build()
+            binding.mapView.getMapboxMap().setCamera(cameraOptions)
+        }
+    }
+
+    private val locationObserver: LocationObserver = object : LocationObserver {
+        override fun onRawLocationChanged(rawLocation: Location) {
+            Log.d(InactiveRouteStylingActivity.TAG, "raw location $rawLocation")
```
[the remark this answers is not in the document]
Comment (cafesilencio) [reply qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/InactiveRouteStylingActivity.kt:155]: done

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/InactiveRouteStylingActivity.kt [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+import com.mapbox.navigation.qa_test_app.utils.Utils
+import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
+import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLine
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineColorResources
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineResources
+
+class InactiveRouteStylingActivity : AppCompatActivity() {
```
[the remark this answers is not in the document]
Comment (cafesilencio) [reply qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/InactiveRouteStylingActivity.kt:38]: fixed

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RouteTrafficUpdateActivity.kt [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLine
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineColorResources
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineResources
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class RouteTrafficUpdateActivity : AppCompatActivity() {
```
[the remark this answers is not in the document]
Comment (cafesilencio) [reply qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RouteTrafficUpdateActivity.kt:25]: fixed

File: qa-test-app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -4,4 +4,7 @@
     <string name="routeline_activity" translatable="false">Internal Development Activity</string>
     <string name="label_start_navigation" translatable="false">Start Navigation</string>
     <string name="route_drawing_utility" translatable="false">Internal Route Drawing Utility</string>
+    <string name="route_restriction_activity_description" translatable="false">Displays a restricted section of a route. \n\n You should see a section near the middle of the route that has a magenta colored, dashed line appearance.</string>
```
[the remark this answers is not in the document]
Comment (cafesilencio) [reply qa-test-app/src/main/res/values/strings.xml:7]: done

File: qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RouteRestrictionsActivity.kt (hunk 1 above)
Comment (cafesilencio) [reply qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RouteRestrictionsActivity.kt:20]: fixed

**Review by Guardiola31337 — APPROVED** [review]
Thanks for addressing the feedback @cafesilencio :shipit:

**Pull request merged**